flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -50,6 +52,7 @@ template <typename T> class DirectiveAttributeVisitor {
     Symbol::Flag defaultDSA{Symbol::Flag::AccShared}; // TODOACC
     std::map<const Symbol *, Symbol::Flag> objectWithDSA;
     bool withinConstruct{false};
+    std::optional<int64_t> maskedTId;
```

Comment:
It appears that we don't really need to store this to check anything, so the `SetMaskedTId` and `GetMaskedTId` functions are not necessary either at this stage. So, If I'm not wrong, these MASKED-related constructs could also just be treated as regular block constructs (no need for special cases in `Pre` and `Post`).

Please do let me know if you don't agree and you had some later uses planned for this.

Reply:
Hi @skatrak , I do agree. I have removed the changes.